client/my-sites/term-tree-selector/terms.jsx

```diff
@@ -94,7 +94,8 @@ const TermTreeSelectorList = React.createClass( {
 		const forceUpdate = (
 			! isEqual( prevProps.selected, this.props.selected ) ||
 			prevProps.loading && ! this.props.loading ||
-			( ! prevProps.terms && this.props.terms )
+			( ! prevProps.terms && this.props.terms ) ||
+			( ( prevProps.terms && this.props.terms ) && ( prevProps.terms.length !== this.props.terms.length ) )
```

Comment:
In `PostSelector`, [we call `recomputeRowHeights` when posts reference has changed](https://github.com/Automattic/wp-calypso/blob/6fabdf8aa234c901d44c620f52c6210d3d77d5af/client/my-sites/post-selector/selector.jsx#L121-L123), which calls `forceUpdate` under the hood ([docs](https://github.com/bvaughn/react-virtualized/blob/master/docs/VirtualScroll.md#recomputerowheights-index-number)). If effective, might be why we haven't encountered the same issue there.